coniferest/session/__init__.py

```diff
@@ -74,8 +74,17 @@ class Session:
     >>> s.run()
     """
     @staticmethod
-    def _validate_callbacks(callbacks):
-        return all([isinstance(cb, Callable) for cb in callbacks])
+    def _prepare_callbacks(input_argument):
+        if input_argument is None:
+            return []
+
+        if isinstance(input_argument, list):
+            callbacks = input_argument
+        else:
+            callbacks = [input_argument,]
```

Comment:
Sorry for being annoying. But there are three different mutually exclusive cases are allowed for input argument, so I suppose that the following would be more clear:

```python
        if input_argument is None:
            callbacks = []
        elif isinstance(input_argument, list):
            callbacks = input_argument
        else:
            callbacks = [input_argument,]
```